src/main/java/ru/spbstu/controller/VcfController.java

```diff
@@ -27,4 +31,22 @@
             return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body("Incorrect vcf file");
         }
     }
+
+    @PostMapping("/getAnnotatedVariants")
+    @ResponseBody
+    public AnnotatedVariantList getAnnotatedVariants(@RequestBody VariantList requestedVariantList) {
+        List<Variant> requestedVariants = requestedVariantList.getVariants();
+        List<Variant> annotatedVariants = new ArrayList<>();
+
+        for (Variant var: requestedVariants) {
+            Optional<Variant> variant = variantService.find(var.getChromosome(), var.getPosition(),
+                    var.getReferenceBase(), var.getAlternateBase());
+            if (variant.isPresent() && variant.get().getAnnotations() != null) {
+                annotatedVariants.add(variant.get());
+            }
+        }
+        AnnotatedVariantList annotatedVariantList = new AnnotatedVariantList();
+        annotatedVariantList.setVariants(annotatedVariants);
```

Comment:
Instead of "create then set" you can use a constructor (or builder)

Reply:
Shortened code by returning result of all arguments constructor execution